### ISSUE TITLE
Writing marble tests

### ISSUE THREAD

**Issue by jmfayard** [RubyLichtenstein/RxKotlinTest#17]
Hello,
what I lack when testing rx java things is the ability to quickly produce input observables.

RxJs has super cool functions to produce those 

`cold("--a--b--#")`: On frame 20 emit a, on frame 50 emit b, and on frame 80, error`

`hot("-a-^-b--|")`: In a hot observable, on frame -20 emit a, then on frame 20 emit b, and on frame 50, complete.

https://github.com/ReactiveX/rxjs/blob/master/doc/writing-marble-tests.md

something like that in kotlin would be super valuable

**Comment by RubyLichtenstein** [comment]
Looks interesting, thank you.

**Comment by RubyLichtenstein** [comment]
Hi @jmfayard 

Do you want to work on this feature?

**Comment by jmfayard** [comment]
@RubyLichtenstein I can contribute what I already did. I fear I don't understand enough rxjava to make it right by myself though

**Comment by RubyLichtenstein** [comment]
Great, so please fork the project and add what you all ready done

What about porting the JavaScript code to kotlin code?

**Comment by jmfayard** [comment]
We should steal their ideas, not their code. A lot of how it's implemented rely on how javascript works. e.g you don't have dynamic types on the JVM.

**Comment by RubyLichtenstein** [comment]
I think we should take what we can from the js code. 
We shouldn't do work that already done
Its not stealing if its open source :)

**Comment by RubyLichtenstein** [comment]
https://github.com/ReactiveX/rxjs/blob/master/src/internal/testing/TestScheduler.ts

**Comment by RubyLichtenstein** [comment]
Most of the code should be the same

**Comment by jmfayard** [comment]
I meant stealing in a positive sense, like T.S. Eliott quote `Good artists copy; great artists steal.`

**Comment by jmfayard** [comment]
https://github.com/RubyLichtenstein/RxTest/pull/19

**Comment by RubyLichtenstein** [comment]
Great job!

I'll review it as soon as I have time

**Comment by RubyLichtenstein** [comment]
So the work you did is not based on the js implementation?

**Comment by jmfayard** [comment]
no, I didn't have the code of the js implementation at that time.
